Source/cmake/FindGLibCompileResources.cmake

```diff
@@ -0,0 +1,68 @@
+pkg_get_variable(GLIB2_PREFIX glib-2.0 prefix)
+find_program(
+    GLIB_COMPILE_RESOURCES_EXE
+    NAMES glib-compile-resources
+    HINTS ${GLIB2_PREFIX}
+    REQUIRED
+)
+execute_process(
+    COMMAND ${GLIB_COMPILE_RESOURCES_EXE} --version
+    OUTPUT_VARIABLE glib_compile_resources_version
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+message(STATUS "Found glib-compile-resources (required): "
+    ${GLIB_COMPILE_RESOURCES_EXE}
+    " (${glib_compile_resources_version})"
+)
+
+set(glib_compile_resources_has_depfile_bug FALSE)
+if ("${glib_compile_resources_version}" VERSION_LESS 2.77)
+    set(glib_compile_resources_has_depfile_bug TRUE)
+endif ()
+
+function(GLIB_COMPILE_RESOURCES)
+    set(zeroArgKeywords "")
+    set(oneArgKeywords OUTPUT SOURCE_XML)
+    set(manyArgsKeywords RESOURCE_DIRS)
+    cmake_parse_arguments(
+        PARSE_ARGV 0 ARG
+        "${zeroArgKeywords}" "${oneArgKeywords}" "${manyArgsKeywords}"
+    )
+
+    file(RELATIVE_PATH rel_output       ${CMAKE_BINARY_DIR} ${ARG_OUTPUT})
+    file(RELATIVE_PATH rel_source_xml   ${CMAKE_BINARY_DIR} ${ARG_SOURCE_XML})
+    set(resource_dir_args "")
+    foreach (abs_resource_dir IN LISTS ARG_RESOURCE_DIRS)
+        file(RELATIVE_PATH rel_resource_dir ${CMAKE_BINARY_DIR} ${abs_resource_dir})
+        list(APPEND resource_dir_args --sourcedir=${rel_resource_dir})
+    endforeach ()
+
+    set(additional_cmd_line)
+    if (${glib_compile_resources_has_depfile_bug})
+        # Workaround for older versions of glib-compile-resources lacking this fix:
+        # https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3460
+        #
+        # Affected versions produce broken depfiles that look like this:
+        #   foo.xml: resource1 resource2
+        # But depfiles should look like this:
+        #   foo.c: foo.xml resource1 resource2
+        set(additional_cmd_line &&
+            perl -pi ${CMAKE_SOURCE_DIR}/Tools/Scripts/fix-glib-resources-depfile.pl
```

Comment:
There is also a `${TOOLS_DIR}` variable, which could be used instead of `${CMAKE_SOURCE_DIR}/Tools`.